src/main/java/net/snowflake/ingest/utils/IcebergDataTypeParser.java

```diff
@@ -44,12 +55,26 @@ public class IcebergDataTypeParser {
   private static final String ELEMENT_REQUIRED = "element-required";
   private static final String VALUE_REQUIRED = "value-required";
 
+  private static final LogicalTypeAnnotation STRING = LogicalTypeAnnotation.stringType();
+  private static final LogicalTypeAnnotation DATE = LogicalTypeAnnotation.dateType();
+  private static final LogicalTypeAnnotation TIME_MICROS =
+      LogicalTypeAnnotation.timeType(
+          false /* not adjusted to UTC */, LogicalTypeAnnotation.TimeUnit.MICROS);
+  private static final LogicalTypeAnnotation TIMESTAMP_MICROS =
+      LogicalTypeAnnotation.timestampType(
+          false /* not adjusted to UTC */, LogicalTypeAnnotation.TimeUnit.MICROS);
+  private static final LogicalTypeAnnotation TIMESTAMPTZ_MICROS =
+      LogicalTypeAnnotation.timestampType(
+          true /* adjusted to UTC */, LogicalTypeAnnotation.TimeUnit.MICROS);
+
+  private static final int DECIMAL_INT32_MAX_DIGITS = 9;
+  private static final int DECIMAL_INT64_MAX_DIGITS = 18;
+  private static final int DECIMAL_MAX_DIGITS = 38;
+  private static final int DECIMAL_MAX_BYTES = 16;
+
   /** Object mapper for this class */
   private static final ObjectMapper MAPPER = new ObjectMapper();
 
```

Comment:
its unclear why you made this change. I see the only difference is in the avro-sanitization of the column name. Is that logic causing issues? 
Copy-pasting code from the iceberg-parquet library will need to consider licensing. IANAL Apache 2.0 requires adding the license as well as a summary of the modifications made while copying their code.

So if we really need to do this, I'd create an explicit TypeToMessageType file, add the proper license attribution at the top, and list what we've modified again in the file header.

My vote is to not do this at all.

If you want to keep the original column name around why not add a new field/property into ParquetTypeInfo, and pass the column name in there from ParquetTypeGenerator.generateColumnParquetTypeInfo ?

Reply:
Reverted.